docs/Driver.md

```diff
@@ -0,0 +1,311 @@
+# The Swift Driver, Compilation Model, and Command-Line Experience
+
+_or, "why can't I only compile the files that changed?"_
+
+The Swift compiler's command-line interface resembles that of other compilers,
+particularly GCC and Clang. However, Swift's compilation model and some of its
+language features make it a bit tricky to plug into a larger build system. In
+particular, there's no correct way to specify a "one command per file" build
+rule for a normal Swift module.
+
+("What?" I know! Read on.)
+
+The target audience for this document is people who want to integrate the Swift
+compiler into their build system, rather than using Xcode or the package
+manager (`swift build`). If you're looking to work on the driver itself...well,
+this is probably still useful to you, but you should also check out
+DriverInternals.rst and maybe DependencyAnalysis.rst as well. If you're just
+using Xcode or SwiftPM and want to find out what mysterious command-line
+options you could be passing, `swiftc --help` is a better choice.
+
+If you're invoking `swift -frontend` directly, and you aren't working on the
+compiler itself...well, this document should convince you to not do that.
+
+Some terms:
+
+- For the purposes of this document, a _module_ is a single distributable unit
+  of API. (We use this term for a lot of other things too, though; check out
+  Lexicon.rst for the full list.) "Foundation" is a single module, as is the
+  Swift standard library ("Swift"). An app is a module too.
+
+- A _compilation unit_ is a set of source files that are compiled together. In
+  Swift, everything intended to be in the same module must be part of the same
+  compilation unit, and every file in the same compilation unit are assumed to
+  be part of the same module. Doing anything else is unsupported.
+
+- The _driver_ is the program that's run when you invoke `swift` or `swiftc`
+  directly. This doesn't actually compile anything itself; instead it invokes
```

Comment:
I found "directly" vague here. I understand the driver to be the program that's run when you invoke these executables *without* using the `-frontend` flag. The driver generates a series of commands that, for the most part, simply invoke `swift -frontend`.

"Directly" to me is unintuitive because when I invoke `swift MyFile.swift`, my understanding is that that's a pretty "direct" invocation. An "indirect" invocation would be when I invoke `xcodebuild` or something.

So, perhaps change the wording here a bit?